From: Christoph Hellwig <hch@lst.de>
Date: Sat, 7 Aug 2010 18:17:56 +0200
Subject: [PATCH] block: remove wrappers for request type/flags
Git-commit: 33659ebbae262228eef4e0fe990f393d1f0ed941
References: FATE#311692
Patch-Mainline: 2.6.36

Remove all the trivial wrappers for the cmd_type and cmd_flags fields in
struct requests.  This allows much easier grepping for different request
types instead of unwinding through macros.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <jaxboe@fusionio.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>

Automatically created from "patches.suse/block-remove-wrappers-for-request-type-flags.patch" by xen-port-patches.py

--- sle11sp2-2011-06-14.orig/drivers/xen/blkfront/blkfront.c	2011-05-05 14:54:11.000000000 +0200
+++ sle11sp2-2011-06-14/drivers/xen/blkfront/blkfront.c	2010-11-22 13:25:54.000000000 +0100
@@ -687,7 +687,7 @@ static int blkif_queue_request(struct re
 
 	ring_req->operation = rq_data_dir(req) ?
 		BLKIF_OP_WRITE : BLKIF_OP_READ;
-	if (blk_barrier_rq(req))
+	if (req->cmd_flags & REQ_HARDBARRIER)
 		ring_req->operation = BLKIF_OP_WRITE_BARRIER;
 
 	ring_req->nr_segments = blk_rq_map_sg(req->q, req, info->sg);
@@ -746,7 +746,7 @@ void do_blkif_request(struct request_que
 
 		blk_start_request(req);
 
-		if (!blk_fs_request(req)) {
+		if (req->cmd_type != REQ_TYPE_FS) {
 			__blk_end_request_all(req, -EIO);
 			continue;
 		}
--- sle11sp2-2011-06-14.orig/drivers/xen/blktap2/device.c	2011-05-31 11:19:16.000000000 +0200
+++ sle11sp2-2011-06-14/drivers/xen/blktap2/device.c	2010-11-22 13:25:54.000000000 +0100
@@ -838,13 +838,13 @@ blktap_device_run_queue(struct blktap *t
 	BTDBG("running queue for %d\n", tap->minor);
 
 	while ((req = blk_peek_request(rq)) != NULL) {
-		if (!blk_fs_request(req)) {
+		if (req->cmd_type != REQ_TYPE_FS) {
 			blk_start_request(req);
 			__blk_end_request_all(req, -EIO);
 			continue;
 		}
 
-		if (blk_barrier_rq(req)) {
+		if (req->cmd_flags & REQ_HARDBARRIER) {
 			blk_start_request(req);
 			__blk_end_request_all(req, -EOPNOTSUPP);
 			continue;
--- sle11sp2-2011-06-14.orig/drivers/xen/blktap2-new/device.c	2011-05-31 11:19:12.000000000 +0200
+++ sle11sp2-2011-06-14/drivers/xen/blktap2-new/device.c	2011-02-24 16:23:08.000000000 +0100
@@ -240,7 +240,7 @@ blktap_device_run_queue(struct blktap *t
 		if (!rq)
 			break;
 
-		if (!blk_fs_request(rq)) {
+		if (rq->cmd_type != REQ_TYPE_FS) {
 			__blktap_end_queued_rq(rq, -EOPNOTSUPP);
 			continue;
 		}

